targets/spidermonkey/templates/register.c

```diff
@@ -117,6 +128,17 @@ void js_register_${generator.prefix}_${current_class.class_name}(JSContext *cx,
 //  JS_SetPropertyAttributes(cx, global, "${current_class.target_class_name}", JSPROP_ENUMERATE | JSPROP_READONLY, &found);
 
     // add the proto and JSClass to the type->js info hash table
+\#if (COCOS2D_VERSION >= 0x00031000)
```

Comment:
there is a function (not sure when it was added) called `js_get_type_from_native<>` that helps with the `TypeClass` thing. Perhaps the code for the non-v3.10 guard could be simplified by calling that function